src/components/dialog/sign-in-dialog/SignInDialog.jsx

```diff
@@ -0,0 +1,111 @@
+import { useState } from "react";
+import PropTypes from "prop-types";
+import { SignInDialogData } from "data";
+import styles from "./SignInDialog.module.css";
+import { DialogActionsCloseIcon } from "components";
+import { EyeCloseIcon, EyeOpenIcon, spinnerGIF } from "assets";
+
+import {
+  Box,
+  Button,
+  Checkbox,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  FormControlLabel,
+  IconButton,
+  InputAdornment,
+  TextField,
+  Typography,
+} from "@mui/material";
+
+const { buttonData } = SignInDialogData;
+
+export const SignInDialog = ({ openSignInDialog, setOpenSignInDialog }) => {
+  const [showPassword, setShowPassword] = useState(false);
+
+  const handleClose = () => {
+    setOpenSignInDialog(false);
+  };
+
+  const handleClickShowPassword = () => {
+    setShowPassword((prevShowPassword) => !prevShowPassword);
+  };
+
```

Comment:
Liked that you are using functions here rather than making the UI code look stuffy